fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/OnlineIndexerConflictsTest.java

```diff
@@ -201,5 +201,60 @@ public void run() {
         assertSame(recordStore.getIndexState(index.getName()), IndexState.READABLE);
         assertEquals(0, timer.getCount(FDBStoreTimer.Events.COMMIT_FAILURE));
     }
+
+    @Test
+    public void testOnlineIndexerIdempotentNested() throws Exception {
+        // If a conflict happens, this test should succeed by retries
+        Random rand = new Random();
+        final long seed = rand.nextInt();
+        LOGGER.info("Random seed is " + seed);
+        rand = new Random(seed); // make rand repeatable (note that multi threading behaviour is not fully repeatable in this test)
+
+        List<TestRecords1Proto.MySimpleRecord> records =
+                LongStream.range(0, 400).mapToObj(val -> TestRecords1Proto.MySimpleRecord.newBuilder().setRecNo(val).setNumValue2((int)val + 1).build()
+                ).collect(Collectors.toList());
+
+        Index index = new Index("newIndex", field("num_value_2"), IndexTypes.VALUE);
+        FDBRecordStoreTestBase.RecordMetaDataHook hook = metaDataBuilder -> metaDataBuilder.addIndex("MySimpleRecord", index);
+        openSimpleMetaData();
+        try (FDBRecordContext context = openContext()) {
+            for (int i = 50; i <= 150; i++) {
+                // even numbers 100-300 (that's 101 values)
+                recordStore.saveRecord(records.get(i * 2));
+            }
+            context.commit();
+        }
+
+        openSimpleMetaData(hook);
+        try (FDBRecordContext context = openContext()) {
+            recordStore.markIndexWriteOnly(index.getName()).join();
+            context.commit();
+        }
+
+        int[] inserts = {99, 101, 133, 201, 350, 299, 301};
+        for (int i = 0; i < inserts.length; i++) {
```

Comment:
I don't think doing this in a loop really adds much. Probably sufficient to just have the one insert. As written, this also builds the full index repeatedly within the loop, which probably isn't quite what we want.

Reply:
Actually that is intentional, I was trying to add a single record each time outside the range (from both sides), next the the first/last items, and in the middle of it. The result is repeatedly adding a record while repeatedly indexing.

Reply:
I believe that the way that `OnlineIndexer.buildRange` works, it'll build the whole range, so I think what this test is currently doing is:

1. The first iteration, building the index and inserting a record in separate transactions and asserting that they both succeed
1. All subsequent iterations, inserting a record and then validating that the index doesn't need to be built because it already has

So, I'm not quite sure it's testing what's desired, unless I'm misunderstanding the goal here.

Reply:
Is there an easy way to delete the index?

Reply:
If you create a new record store and call `clearAndMarkIndexWriteOnly`, that will delete all of the index's data, if that's what you want to do.

Reply:
Got it: 
        FDBRecordStoreTestBase.RecordMetaDataHook hookDel = metaDataBuilder -> metaDataBuilder.removeIndex(index.getName());

Reply:
Sorry I didn't see your reply before submitting (my browser, as it seems, does not refresh frequently enough). Shall I change the removeIndex to clearAndMarkIndexWriteOnly?

Reply:
It depends on what you are trying to accomplish. The `removeIndex` method will...remove the index from the meta-data (so it will require adding it back to then build the index, if that's the idea). Whereas the method I gave simply "clears out" the index. I _think_ you just want to clear it out here, but I could be misunderstanding the desired effect.

Reply:
Either one will work for my desired affect (which is to add a record in different locations while indexing). Would you be willing to review it as is?